tools/wake/main.cpp

```diff
@@ -117,9 +117,21 @@ std::string globish_to_like(const std::string &str) {
   return glob;
 }
 
-std::string globish_to_like(const char *str) {
-  std::string glob(str);
-  return globish_to_like(glob);
+template <class QueryF>
+Set<long> apply_inspection_query(std::unordered_map<long, JobReflection> &captured_jobs,
+                                 Set<long> current,
+                                 const std::vector<std::vector<std::string>> &query, QueryF qf) {
+  for (const std::vector<std::string> &and_part : query) {
+    std::vector<JobReflection> hits = {};
+    for (const std::string &or_part : and_part) {
+      if (or_part == "") continue;
```

Comment:
Why is this here?

Reply:
small optimization that allows for trailing comma.

`wake --job "1,2,3,"`  splits into `["1", "2", "3", ""]`

Reply:
But now you can't match on the empty string because of it. It changes the semantics.

Reply:
Use case: `wake --label ""` to find all the jobs that people did not add labels to

Reply:
hmm fair. I don't feel strongly towards this and it does change the semantics